app/services/imports/case_logs_import_service.rb

```diff
@@ -0,0 +1,278 @@
+module Imports
+  class CaseLogsImportService < ImportService
+    def create_logs(folder)
+      import_from(folder, :create_log)
+    end
+
+  private
+
+    GN_SH = {
+      general_needs: 1,
+      supported_housing: 2,
+    }.freeze
+
+    SR_AR_IR = {
+      social_rent: 1,
+      affordable_rent: 2,
+      intermediate_rent: 3,
+    }.freeze
+
+    # For providertype, values are reversed!!!
+    PRP_LA = {
+      private_registered_provider: 1,
+      local_authority: 2,
+    }.freeze
+
+    IRPRODUCT = {
+      rent_to_buy: 1,
+      london_living_rent: 2,
+      other_intermediate_rent_product: 3,
+    }.freeze
+
+    # These must match our form
+    RENT_TYPE = {
+      social_rent: 0,
+      affordable_rent: 1,
+      london_affordable_rent: 2,
+      rent_to_buy: 3,
+      london_living_rent: 4,
+      other_intermediate_rent_product: 5,
+    }.freeze
+
+    # Order matters since we derive from previous values (uses attributes)
+    def create_log(xml_doc)
+      attributes = {}
+
+      # Required fields for status complete or logic to work
+      attributes["startdate"] = start_date(xml_doc)
+      attributes["owning_organisation_id"] = find_organisation_id(xml_doc, "OWNINGORGID")
+      attributes["managing_organisation_id"] = find_organisation_id(xml_doc, "MANINGORGID")
+      attributes["previous_postcode_known"] = previous_postcode_known(xml_doc)
+      attributes["ppostcode_full"] = previous_postcode(xml_doc, attributes)
+      attributes["needstype"] = needs_type(xml_doc)
+      attributes["lar"] = london_affordable_rent(xml_doc)
+      attributes["irproduct"] = unsafe_string_as_integer(xml_doc, "IRPRODUCT")
+      attributes["irproduct_other"] = field_value(xml_doc, "xmlns", "IRPRODUCTOTHER")
+      attributes["rent_type"] = rent_type(xml_doc, attributes)
+      attributes["rsnvac"] = unsafe_string_as_integer(xml_doc, "Q27")
+      attributes["renewal"] = renewal(attributes)
+      (1..8).each do |index|
+        attributes["age#{index}"] = age(xml_doc, index)
+        attributes["sex#{index}"] = sex(xml_doc, index)
+        attributes["ecstat#{index}"] = unsafe_string_as_integer(xml_doc, "P#{index}Eco")
+      end
+      # attributes["hhmemb"] =
+
+      # Not specific to our form but required for CDS and can't be inferred
+      # attributes["form"] = Integer(field_value(xml_doc, "xmlns", "FORM"))
+      # attributes["lettype"] = let_type(xml_doc, attributes)
+
+      case_log = CaseLog.new(attributes)
+      case_log.save!
```

Comment:
Might be quicker (if we're importing thousands of cases), to create CaseLogs in memory (could be parallelized) and then `upsert` them all at once? (Can batched if all at once is too many)?

Reply:
Yup, I was planing to look at performance and robustness once we got all the fields in ;)